packages/@aws-cdk/aws-rds/test/cluster.test.ts

```diff
@@ -760,7 +760,55 @@ describe('cluster', () => {
     cluster.addRotationSingleUser();
 
     // THEN
-    expect(() => cluster.addRotationSingleUser()).toThrow(/A single user rotation was already added to this cluster/);
+    expect(() => cluster.addRotationSingleUser()).toThrow(/A single user rotation for master secret is already added to this cluster/);
+
+
+  });
+
+  test('throws when trying to add single user rotation for additional secret which is master secret', () => {
+    // GIVEN
+    const stack = new cdk.Stack();
+    const vpc = new ec2.Vpc(stack, 'VPC');
+    const cluster = new DatabaseCluster(stack, 'Database', {
+      engine: DatabaseClusterEngine.AURORA_MYSQL,
+      credentials: { username: 'admin' },
+      instanceProps: {
+        instanceType: ec2.InstanceType.of(ec2.InstanceClass.BURSTABLE2, ec2.InstanceSize.SMALL),
+        vpc,
+      },
+    });
+
+    // WHEN
+    cluster.addRotationSingleUser();
+
+    // THEN
+    expect(() => cluster.addRotationSingleUser({ secret: cluster.secret!! })).toThrow(/A single user rotation for master secret is already added to this cluster/);
```

Comment:
1. No need for the `!!` here.
2. Please break this into multiple lines:

```suggestion
    expect(() => {
      cluster.addRotationSingleUser({ secret: cluster.secret });
    }).toThrow(/A single user rotation for master secret is already added to this cluster/);
```